src/common/vector/value_vector.cpp

```diff
@@ -87,6 +87,9 @@ uint32_t ValueVector::getDataTypeSize(const LogicalType& type) {
     case LogicalTypeID::VAR_LIST: {
         return sizeof(list_entry_t);
     }
+    case LogicalTypeID::ARROW_COLUMN: {
```

Comment:
I still don't think this is a good idea. It doesn't make sense to me to have ARROW column/array as a logical type.